### PR TITLE
[WIP] Optimize prow loader with COPY protocol, bulk operations, and GCS improvements

### PR DESCRIPTION
**Pull request by mstaeble**
## Summary

- Replaces per-row DB operations (findOrAddTest, CreateInBatches, per-row ProwJob upserts) with bulk PostgreSQL operations using COPY protocol, temp tables, and writable CTEs
- Replaces 800k+ in-memory job run ID cache with a temp table COPY + LEFT JOIN anti-join to find new runs
- Removes all shared mutable caches and their locks (prowJobRunTestCache, suiteCache, prowJobCacheLock, prowJobRunCacheLock)
- Pre-filters candidates before the goroutine loop so workers only receive new runs
- Optimizes GCS artifact fetching with server-side MatchGlob filtering, SetAttrSelection, and removal of unnecessary Attrs/Generation pinning
- Increases worker concurrency from 10 to 50
- Pre-compiles release config regexes at construction

## Behavioral change

`extractTestCases` now attributes tests to their actual child suite name rather than inheriting the parent suite's ID. The old behavior assigned all nested JUnit tests to the top-level suite, which was incorrect for tests in child suites with different names.

## Benchmarks (staging-2, all releases, 4h lookback)

| Metric | Baseline | Optimized |
|---|---|---|
| Jobs imported | ~1,100 | ~830 |
| **Import time** | **5m11s** | **38s** |
| Speedup | | **8x** |

Job count varies by run (different 4-hour windows). Per-job throughput is consistently much higher.

## Test plan

- [x] `gofmt`, `go vet`, `golangci-lint` pass
- [x] `make test` passes
- [x] Benchmarked against staging-2 database with production-scale data
- [x] Verified row counts and test outputs match
- [x] Independent code review (two rounds) found and fixed critical MatchGlob bug, ctx-cancellation edge case, correlated subquery, and string-concat VALUES clause

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved bulk loading of job-run test results for higher throughput and faster processing.

* **Bug Fixes**
  * Job loading now excludes Prow jobs in non-terminal states (pending/triggered).
  * Enhanced glob-based discovery for events, intervals/timelines, cluster-data, and JUnit artifacts, with more consistent behavior when files are missing.

* **Tests**
  * Added unit coverage for test-case extraction and suite naming/status handling.

* **Documentation**
  * Not included in this change.

### CODE REVIEW

**Comment by mstaeble** [comment]
@coderabbitai review

**Comment by mstaeble** [comment]
@coderabbitai full review

**Comment by mstaeble** [comment]
@coderabbitai full review

File: pkg/dataloader/prowloader/prow.go [hunk 1]
```diff
@@ -598,69 +583,174 @@ ORDER BY
...
+	var newIDs []uint
+	for rows.Next() {
+		var id uint
+		if err := rows.Scan(&id); err == nil {
+			newIDs = append(newIDs, id)
+		}
+	}
+	rows.Close()
+
+	return sets.New(newIDs...), nil
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply pkg/dataloader/prowloader/prow.go:749]: Fixed. `rows.Scan` errors now return immediately, and `rows.Err()` is checked after iteration.

File: pkg/dataloader/prowloader/prow.go [hunk 2]
```diff
@@ -1285,151 +1445,110 @@ func gatherLabelsBatch(ctx context.Context, bqClient *bqcachedclient.Client, bui
...
 
-	suiteID := pl.findSuite(syntheticSuite.Name)
-	if suiteID == nil {
-		// this shouldn't happen but if it does we want to know
-		panic("synthetic suite is missing from the database")
+	if !db.IsSuiteImportable(syntheticSuite.Name) {
+		panic("synthetic suite is missing from the importable list")
 	}
-	pl.extractTestCases(syntheticSuite, suiteID, testCases, prowJobRelease, pj.Status.StartTime)
+	extractTestCases(syntheticSuite, testCases)
```
[the remark this answers is not in the document]
Comment (mstaeble) [reply pkg/dataloader/prowloader/prow.go:1450]: Fixed. Replaced `panic()` with return `nil, 0, "", fmt.Errorf(...)` so a missing synthetic suite fails the individual job run rather than crashing the loader.

**Comment by mstaeble** [comment]
@coderabbitai pause
